src/rpcblockchain.cpp

```diff
@@ -5246,3 +5246,167 @@ Value getcheckpoint(const Array& params, bool fHelp)
     return result;
 }
 
+// Brod
+static bool compare_second(const pair<std::string, long>  &p1, const pair<std::string, long> &p2)
+{
+    return p1.second > p2.second;
+}
+json_spirit::Value rpc_getblockstats(const json_spirit::Array& params, bool fHelp)
+{
+    if(fHelp || params.size() < 1 || params.size() > 3 )
+        throw runtime_error(
+            "getblockstats mode [startheight [endheight]]\n"
+            "Show stats on what wallets and cpids staked recent blocks.\n");
+    long mode= std::stol(params[0].get_str());
+    (void)mode; //TODO
+    long lowheight= 0;
+    long highheight= INT_MAX;
+    if(params.size()>=2)
+        lowheight= std::stol(params[1].get_str());
+    if(params.size()>=3)
+        highheight= std::stol(params[2].get_str());
+    CBlockIndex* cur;
+    Object result1;
+    {
+        LOCK(cs_main);
+        cur= pindexBest;
+    }
+    long blockcount = 0;
+    long transactioncount = 0;
+    std::map<int,long> c_blockversion;
+    std::map<std::string,long> c_version;
+    std::map<std::string,long> c_cpid;
+    std::map<std::string,long> c_org;
+    long researchcount = 0;
+    double researchtotal = 0;
+    double interesttotal = 0;
+    int64_t minttotal = 0;
+    int64_t stakeinputtotal = 0;
+    long poscount = 0;
+    long emptyblockscount = 0;
+    long l_first = INT_MAX;
+    long l_last = 0;
+    int l_first_time,l_last_time;
+    for( ; (cur
+            &&( cur->nHeight>=lowheight )
+            &&( lowheight>0 || blockcount<=14000 )
+        );
+        cur= cur->pprev
+        )
+    {
+        if(cur->nHeight>highheight)
+            continue;
+        if(l_first>cur->nHeight)
+        {
+            l_first=cur->nHeight;
+            l_first_time=cur->nTime;
+        }
+        if(l_last<cur->nHeight)
+        {
+            l_last=cur->nHeight;
+            l_last_time=cur->nTime;
+        }
+        blockcount++;
+        CBlock block;
+        if(!block.ReadFromDisk(cur->nFile,cur->nBlockPos,true))
+            throw runtime_error("failed to read block");
+        assert(block.vtx.size() > 0);
+        unsigned txcountinblock = 0;
+        if(block.vtx.size()>=2)
+        {
+            txcountinblock+=block.vtx.size()-2;
+            if(block.vtx[1].IsCoinStake())
+            {
+                poscount++;
+                //stakeinputtotal+=block.vtx[1].vin[0].nValue;
+            }
+            else
+                txcountinblock+=1;
+        }
+        transactioncount+=txcountinblock;
+        emptyblockscount+=(txcountinblock==0);
+        c_blockversion[block.nVersion]++;
+        MiningCPID bb = DeserializeBoincBlock(block.vtx[0].hashBoinc,block.nVersion);
```

Comment:
Fixed. The block version parameter was added by me in 4f952e1, which was not merged here.
Should I cherry-pick that commit out of stakev8 too?